### ISSUE TITLE
Make VSCode window transparent

### ISSUE THREAD

**Issue by pcgeek86** [microsoft/vscode#12578]
- VSCode Version: 1.5.3
- OS Version: Windows 10 Anniversary Update
## Feature Request

It would be great if we could make the VSCode window transparent, using a percentage.

This was discussed on StackOverflow back in June.

http://stackoverflow.com/questions/38031369/how-to-make-ms-visual-code-transparent

Cheers,
**Trevor Sullivan**
Docker Captain
Microsoft MVP: Cloud & Data Center Management
https://trevorsullivan.net
https://twitter.com/pcgeek86

**Comment by bpasero** [comment]
This might require Electron API, thereby setting to upstream.

**Comment by alexkrolick** [comment]
Looks like the "vibrancy" setting for macOS would cover this on Electron: https://github.com/electron/electron/blob/master/docs/api/browser-window.md#winsetvibrancytype-macos

Support for it was just added to the Hyper terminal app that also uses Electron: https://github.com/zeit/hyper/releases/tag/1.0.0

**Comment by BobbyBabes** [comment]
What I do now is double-click the title bar to roll-up the VSCode window so that just its title bar is visible. But that becomes annoying because I have to take my hands off the keyboard.
Similar to this (the VScode window is rolled up so that only its title bar is visible). :
![vscode-window-transparent_20161228200840](https://cloud.githubusercontent.com/assets/22870070/21532698/e9ec9b4e-cd52-11e6-823b-b7dd0b0d10b1.png)



So I was looking into this a few hours ago. To make an extension to press and hold a key on the keyboard that makes the VSCode window completely transparent so that I can see the 2 - 3 browser windows (or whatever) under it.

In VSCode I did ``Help >> Toggle Developer Tools``.
Then selected the ``html`` element and added ``display: none;`` to it. I ended up with what's shown in the screenshot below. I just can't get the last bit transparent. Looks like that's the "shell" VSCode is running in.

![vscode-window-transparent_20161228195920](https://cloud.githubusercontent.com/assets/22870070/21532623/3606b25e-cd52-11e6-9173-63c9f899a487.png)

**Comment by PhongAthena** [comment]
I hope this feature will appear soon

**Comment by ohtangza** [comment]
This feature is very useful for running auto-test behind the window during TDD.

**Comment by jonathanphz** [comment]
I've been dying to have this functionality - this would help build UI's without constantly switching apps.

**Comment by thiagomajesk** [comment]
Definitely a vs code must have. That way we could achieve this: https://github.com/Microsoft/vscode/issues/32257

**Comment by jonathanphz** [comment]
Jonathan Perez liked your email
Spark by Readdle

**Comment by isalew** [comment]
Would love the same for the Mac OS as well.

**Comment by paracha3** [comment]
+1

**Comment by webdevbrian** [comment]
👍 looks like someone made a transparent plugin for windows, but mac is still yet to get a solution. https://marketplace.visualstudio.com/items?itemName=s-nlf-fh.glassit

**Comment by Tekbr** [comment]
-- Sorry for the English, I used Google Translator --

Experiencing the extension. 😄  Thanks @webdevbrian . 👍

**Comment by MikePaer** [comment]
Confirming that the "Glassit-VSC" extension for Visual Studio Code (on Windows) works great. 
![2017-10-12 14_17_45-](https://user-images.githubusercontent.com/24686713/31519768-4a181510-af58-11e7-8668-389f249d1c0c.png)

**Comment by FocusedBit** [comment]
@MikePaer I really hope that becomes available for MacOs ... currently not working yet :)

**Comment by darkguy2008** [comment]
+1 We SOOOOOOOO need this in OSX.

**Comment by lucasqueiroz** [comment]
Maybe even an option to set only the explorer and not the code part too?
+1

**Comment by sonhanguyen** [comment]
for OSX user AFloat which has a transparency option does work with vscode, only gotcha is at some point the AFloat menu disappear so you have to turn it on when vscode starts

**Comment by okinskas** [comment]
Is anyone aware of a solution for this on linux?

**Comment by SamHH** [comment]
@okinskas You can give VSCode full window transparency on Linux with [devilspie](https://wiki.archlinux.org/index.php/Per-application_transparency) under Xorg. My `~/.devilspie/opacity.ds` looks like this:

```
( if
( contains ( window_class ) "Code" )
( begin
( spawn_async (str "transset-df -i " (window_xid) " 0.95" ))
)
)
```

**Comment by okinskas** [comment]
@SamHH Works great thanks!

**Comment by lucasqueiroz** [comment]
@SamHH that didn't work for me :/ Do I have to restart the computer or something? I ran `devilspie -a` and when I open VSCode, it says `Set Property to 0.95`, but nothing changes...

**Comment by SamHH** [comment]
@lucasqueiroz Check the link in my comment (Arch Wiki is an _outstanding_ resource for Linux), are you running a compositor? I for example am running Compton.

**Comment by sergeysova** [comment]
Does this issue is a duplicate of #32257?

**Comment by ZenMilan** [comment]
Will this feature be available for linux users, eagerly waiting for the feature.

**Comment by Huyhd** [comment]
Very nice, the Glassit extension works beautifully on Windows 10.

**Comment by Marahin** [comment]
MacOS here, this is a feature I'm waiting for. 👍

**Comment by createdbyjurand** [comment]
Still waiting for native support of transparency in vscode :)
Options:
- background transparency (without text - because the most important thing here is codding, and to make this user friendly we definitely need to see the code best way possible in the first place, and than have fun using vscode environment - and, of course, for different parts of the app like: terminal, explorer, menu, side menu...)
- text transparency (i guess there will be some that will use this also)
- vibrancy effect (this will not work everywhere, however why not let those who are software ready, just have it)

**Comment by raylee** [comment]
TextMate2 allows for RGBA specification in their themes. I think that would be a nice way to handle this in VSCode once support lands upstream, allowing theme authors do so whatever they like for styling the alpha values along with everything else.

In the meantime I'm using afloat on OSX. It's a bit of a kludge as afloat's menus disappear when VSCode changes the menu structure, but you can engage the transparency when VSCode first starts up.

**Comment by DAVIDhaker** [comment]
Will good if VSC will have transparency feature, like gnome-terminal. Transparent only background, text is not transparent.

KDE

**Comment by jumpyoshim** [comment]
+1 for adding this to OSX.

**Comment by lunarfyre7** [comment]
I modified the code @SamHH to trigger the blur effect in kde, it yields a very nice result with the new blurring in plasma 5.13! I found that for some reason transset-df wasn't working on my system so I used a kwin rule instead to set the opacity.
```
( if
( contains ( window_class ) "Code" )
( begin
( spawn_async (str "xprop -f _KDE_NET_WM_BLUR_BEHIND_REGION 32c -set _KDE_NET_WM_BLUR_BEHIND_REGION 0 -id " (window_xid) ))
)
)
```
![screenshot_20180709_135708](https://user-images.githubusercontent.com/15057106/42475802-9f5443d0-8380-11e8-818c-f5b088b62579.png)
![screenshot_20180709_135907](https://user-images.githubusercontent.com/15057106/42475807-a1185cba-8380-11e8-8c43-80bcb8c3f141.png)

**Comment by LeviZell** [comment]
For linux users, I found this solutions:

https://jwigley.com/2012/04/27/making-ubuntu-unity-look-beautiful-by-enabling-transparency/

Needed a restart but works perfectly for ubuntu 16.04

**Comment by lloydjatkinson** [comment]
Can we have just a terminal transparent?

**Comment by lunarfyre7** [comment]
@lloydjatkinson that might be possible with custom css, at least for it to show the editor underneath.

**Comment by IdiotLeon** [comment]
MacOS here too!
Desperately need this feature!

**Comment by longdeqidao** [comment]
NEED THIS LIKE iTerm2.

**Comment by erasmo-marin** [comment]
OSX here, transparent backgrounds are very common in Mac OS, even system apps like finder use it. Please do it ❤

**Comment by raylee** [comment]
An implementation request for anyone who wants to work on this feature: Textmate allows RGBA colors in theme files, which permits each part of the interface to have its own opacity values. (Sublime Text 3 also allows this, but last time I tried it ignores the alpha values.)

In the meantime on OSX I'm using this fork of afloat, which contains a fix I made to delay installing the extra menu items related to afloat's features, which makes it work in more situations than before: https://github.com/w0lfschild/afloat

It's not perfect by any means. Afloat is a bit annoying to install on OSX due to System Integrity Protection, and if the app rewrites the menus the afloat plugin doesn't notice and so transparency controls will be gone until the app is restarted.

Supporting transparency directly in the editor would really help. I find transparent backgrounds to be an accessibility issue for me -- my eyes are much happier with a faint background showing through and I end the day feeling less eye strain. Also it's pretty :-)

**Comment by ikbelkirasan** [comment]
This devilspie script should work for ubuntu 18.04 users.

```
( if
( contains ( window_class ) "Code" )
( begin
( spawn_async (str "xprop -id " (window_xid) " -f _NET_WM_WINDOW_OPACITY 32c -set _NET_WM_WINDOW_OPACITY 0xdfffffff") )
)
)
```

**Comment by LutfiZakaria** [comment]
Thanks to @ikbelkirasan . It works very nice in Ubuntu 18.04. 
I have -set _NET_WM_WINDOW_OPACITY 0xd0000000 rather than -set _NET_WM_WINDOW_OPACITY 0xdfffffff.  My code screen looks very clear now
![screenshot from 2018-10-26 05-49-12](https://user-images.githubusercontent.com/43399302/47534957-8d06a980-d8e3-11e8-8a5d-e3bea24d3c68.png)
![screenshot from 2018-10-26 05-49-12](https://user-images.githubusercontent.com/43399302/47534970-9859d500-d8e3-11e8-9cc6-43d41a1d949b.png)

**Comment by loretoparisi** [comment]
Please support this for macOS 🤞

**Comment by a-marionette** [comment]
+1 for macOS support

**Comment by webispy** [comment]
Thanks to @SamHH 
Here is my vscode screenshot on a Linux machine. KDE plasma (blur) + devilspie + vscode

![vscode_blur](https://user-images.githubusercontent.com/1968467/49569286-ddf3ce00-f976-11e8-9c3d-fecf89f461db.png)

And, following is my `~/.devilspie/vscode.ds` script.
```
( if
( contains ( window_class ) "Code" )
( begin
( spawn_async (str "xprop -id " (window_xid) " -f _KDE_NET_WM_BLUR_BEHIND_REGION 32c -set _KDE_NET_WM_BLUR_BEHIND_REGION 0 ") )
( spawn_async (str "xprop -id " (window_xid) " -f _NET_WM_WINDOW_OPACITY 32c -set _NET_WM_WINDOW_OPACITY 0xdfffffff") )
)
)
```

**Comment by Superpencil** [comment]
This is a very basic PR that enables vibrancy on macOS Mojave: https://github.com/Microsoft/vscode/pull/65215 😃

**Comment by BrianD91** [comment]
@webispy @lunarfyre7 Any ideas why I can't seem to get it working? Technically I'm using VSCodium which is a FOSS fork of VSCode, but other than that I don't see why it isn't working for me. Using KDE Neon

**Comment by webispy** [comment]
@BrianD91 Please check your KDE configuration.
https://www.reddit.com/r/unixporn/comments/9un99u/kde_plasma_do_you_need_so_more_blur/

**Comment by vkubre** [comment]
You can achieve this with compton, just add 
opacity-rule = [
 "94:class_g = 'Code'"
]

**Comment by SoptikHa2** [comment]
@vkubre 

It looks like I need to start compton after code is already running, it doesn't seem to work otherwise. For example: `start code; then start compton; (code is transparent); close code; start code; (code is not transparent)`.

**Comment by fffredy** [comment]
Hi, man I am a Mac user and I don't have enough information about using electron or other things. I downloaded vs code today. And I want to make vs code background transparent as u did in the post. I have already tried to access src/vs/ folder but unfortunately I did not be able to access that. There is not any video for making it transparent. Since I don't know how to re-build vs code I did not be able to make vs code transparent. PLEASE, PLEASE make a video tutorial from beginner to the end on how to make vs code transparent. It looks easy for you to make it but I am sure that it is really hard for a beginner programmers like me. I hope you will reply to me. Have a nice day

**Comment by mafflerbach** [comment]
> You can achieve this with compton, just add
> opacity-rule = [
> "94:class_g = 'Code'"
> ]
> Heres result

Here is for code insider user:

```
opacity-rule =
[
    "70:class_g = 'Code - Insiders'"
];
```

**Comment by StarrKiss** [comment]
The devilspie solutions work, but how do I change the opacity to 51% instead of the given amount using the hex code?

**Comment by ShayBox** [comment]
I would look for a more proper implementation, as just telling your compositor to apply transparency and blur applies to all elements equally, including fonts, this needs to be an internal thing, but that screen shot looks good :+1:

**Comment by BobbyBabes** [comment]
@StarrKiss 
>but how do I change the opacity to 51% instead

[https://dev.to/emmanuelnk/how-to-be-cool-and-make-vscode-transparent-56ib](https://dev.to/emmanuelnk/how-to-be-cool-and-make-vscode-transparent-56ib)

**Comment by StarrKiss** [comment]
@BobbyBabes  No, I got that. For blur on KDE, it only works with hex code. How to I change the hex code value to 51? I tried hex code converters and replacing the numbers at the beginning and end and nothing fixes it.

**Comment by kubre** [comment]
update with compton: Only having transparency with compton sometimes makes code harder to read. You can use [This fork](https://github.com/tryone144/compton) to use blur options also.
Material theme:
![Uploading 2019-06-17_08-29-53.jpg…](Material theme)

1984 Theme:
![Uploading 2019-06-17_08-32-48.jpg…](1984 Theme)

**Comment by BobbyBabes** [comment]
> @BobbyBabes No, I got that. For blur on KDE, it only works with hex code. How to I change the hex code value to 51? I tried hex code converters and replacing the numbers at the beginning and end and nothing fixes it.

It's in the article. The writer's example for 87% :
> -set _NET_WM_WINDOW_OPACITY $(printf 0x%x $((0xffffffff * 87 / 100))

So for 51% it becomes :
`-set _NET_WM_WINDOW_OPACITY $(printf 0x%x $((0xffffffff * 51 / 100))`

**Comment by liuchong** [comment]
macOS: 喵喵喵？

**Comment by WillPower3309** [comment]
Check out https://github.com/coler706/vscode
<img width="1368" alt="screenshot2" src="https://user-images.githubusercontent.com/13266308/63975548-46db2300-ca7d-11e9-8f8b-78c14f73d7b2.png">

**Comment by marcel-dempers** [comment]
If you are running Linux, I found devilspie very useful and made a video on it.
Also found a little formula that helps set a % transparency

Hope this helps!
https://youtu.be/PzObHq72Vug

**Comment by annymosse** [comment]
@marcel-dempers it's not a good solution which we can't customize the vs code part separated such as these pictures [vibrant window theme](https://github.com/microsoft/vscode/issues/32257)

**Comment by juanccamachob94** [comment]
> If you are running Linux, I found devilspie very useful and made a video on it.
> Also found a little formula that helps set a % transparency
> 
> Hope this helps!
> https://youtu.be/PzObHq72Vug

Works on Elementary O.S. 5.0 Juno

**Comment by tiritto** [comment]
Using devilspie or compton transparency is just a big workaround that results in applying same level of transparency to everything all at once which is probably far from desired for many people asking for opacity support. We need a way to make only a specific element transparent without applying it to other elements. For example, you might want to have your VSCode background to be fully transparent without affecting your code/text in the process.

**Comment by bvibhav** [comment]
I've set keyboard shortcuts for these two commands in Ubuntu for making any overall window transparent. However, I would like to control transparency of individual areas in VSCode interface. 

`transset --inc .05 -a`
`transset --dec .05 -a -m .6`

**Comment by vegerot** [comment]
> It looks like it'll be coming into official build!
> 
> #52707

[One year later](https://assets.rbl.ms/16971403/origin.png)

**Comment by Fmstrat** [comment]
Hah, and I just released a GlassIt for Linux: https://marketplace.visualstudio.com/items?itemName=nowsci.glassit-linux

**Comment by annymosse** [comment]
@Fmstrat it make the whole window transparent which is able to get that effect without your ext ,the main goal is to make some part transparent  https://github.com/microsoft/vscode/issues/32257 or [image preview](https://user-images.githubusercontent.com/18223213/29234148-e3a05c2c-7ea9-11e7-95d6-67a73ae6e8b3.png)

**Comment by Fmstrat** [comment]
@annymosse Oh yes, having native fade support is way better but MS has stated they won't merge in Electron focused changes.

@vegerot It looks like that PR was closed and will not be merged.

**Comment by annymosse** [comment]
@Fmstrat look like MS team still against Linux users and won't give up, to make the users switch back to windows , the truth is "WE'RE FREE TO USE ANY SYSTEM WE WOULD LIKE" the freedom of ower minds making this world unlimited, WINDOWS & MAC achieved them end-life and not obligation to waste money for uncomfortable system.

**Comment by darkguy2008** [comment]
> @Fmstrat look like MS team still against Linux users and won't give up, to make the users switch back to windows , the truth is "WE'RE FREE TO USE ANY SYSTEM WE WOULD LIKE" the freedom of ower minds making this world unlimited, WINDOWS & MAC achieved them end-life and not obligation to waste money for uncomfortable system.

Well, nobody is pointing a gun at you to use VSCode on Linux. With plenty of other editors since it's all open-source and all thar jazz, you're free to not to use VSCode then, and leave us Windows users happy with such a great OS.

**Comment by annymosse** [comment]
look like Windows is your house and them team is your family ,then i didn't hurt them after all this's my vision and i have the freedom to explain my opinions under the respect , MS & Apple are commercial companies as they have the freedom to advertise them products (even with some fake less service as they provide) as we have a right to say the our bad experience.
Sorry to the Windows users "I'm not pointing a gun over your heads"  and Linux not my company and all the people know Linux is a free & non-profit organisation .
@darkguy2008 be a white heart heart no matter your skin color is,we're here to make a good world & simple life by our minds not our force and guns.

**Comment by monikaprajpt** [comment]
Hey! 
I use Arch Linux where I have been using picom for transparency and blur effect. It works for vs studio/code oss. I heard compton-tryone provides kawase blur which is not available on picom at the moment.

**Comment by annymosse** [comment]
@monikaprajpt Picom is a fork of compton, which is a fork of xcompmgr-dana, which in turn is a fork of xcompmgr.
all of them affect the process , and you can't customize each part of the editor like: ![transparent image effect](https://user-images.githubusercontent.com/18223213/29234148-e3a05c2c-7ea9-11e7-95d6-67a73ae6e8b3.png)
then the result of these hacks make the text code not readable like this image ![transparent and vibrant](https://user-images.githubusercontent.com/18104031/62838423-2c7f0800-bc7c-11e9-8489-f42e5e815e16.png) because these solutions affect the process instead the elements .

**Comment by gileli121** [comment]
This is a generic solution for Windows 10 version 2004 and above with NVIDIA GPU with CUDA

![image](https://user-images.githubusercontent.com/17680514/82048535-8a35fa00-96bd-11ea-94f9-a46dd0e83697.png)

![image](https://user-images.githubusercontent.com/17680514/82048586-9a4dd980-96bd-11ea-8fbf-98ff3df5580e.png)

![image](https://user-images.githubusercontent.com/17680514/82048622-a76ac880-96bd-11ea-8550-a6c17a42b413.png)

The software that does it is WindowTop and it has an exclusive algorithm that works on any window. No need to write a custom plugin for an editor or modify it. it just works on any text editor using a generic algorithm that powers by NVIDIA GPUs

https://github.com/gileli121/WindowTop/releases/tag/v3.4.5

More info and instructions here:
https://windowtop.info/2020/05/07/how-to-make-your-ide-or-code-editor-with-transparent-background/

Note: this is a paid solution.

**Comment by PascalSenn** [comment]
@monika-prajapati FYI: https://github.com/yshui/picom/pull/382 @tryone1144 is working on the integration for picom

**Comment by WillPower3309** [comment]
> 
> 
> @monika-prajapati FYI: [yshui/picom#382](https://github.com/yshui/picom/pull/382) @tryone144 is working on the integration for picom

This will blur it, but afaik still requires transparency on the vscode end

**Comment by Boss-U** [comment]
Really wish we could adjust each elements transparency in Arch...

**Comment by ckissane** [comment]
@Boss-U My fork works perfectly fine on arch with KDE plasma kwin and other blurring compositors!
https://github.com/ckissane/vscode

**Comment by Fmstrat** [comment]
@Boss-U @ckissane Yikes:
```
This branch is 10 commits ahead, 8736 commits behind microsoft:master. 
```

**Comment by ckissane** [comment]
Lol yeah I'll merge vscode master in

**Comment by ckissane** [comment]
@Fmstrat @Boss-U  now it's 12 commit's ahead 0 behind

**Comment by Minionbob2019** [comment]
Hi all,
I'm not sure if it would at all be helpful but just wanted to suggest this lightweight utility called WndowTop to Windows10 users. (https://windowtop.info/) Give it a try and it can make any code/text editor window (including vscode) transparent in no time. It costs $5.99 for three pcs. I know it's not the same as having this functionality inbuild in vscode but it works well for me on Wondows10. So, I thought of sharing it.

**Comment by jaweii** [comment]
REOPEN THE  FEAQURE REQUEST PLEASE!!!

**Comment by darkguy2008** [comment]
> This issue is being closed to keep the number of issues in our inbox on a manageable level, we are closing issues that are not going to be addressed in the foreseeable future: We look at the number of votes the issue has received and the number of duplicate issues filed. More details [here](https://aka.ms/vscode-out-of-scope). If you disagree and feel that this issue is crucial: We are happy to listen and to reconsider.
> 
> If you wonder what we are up to, please see our [roadmap](https://aka.ms/vscoderoadmap) and [issue reporting](https://aka.ms/vscodeissuereporting) guidelines.
> 
> Thanks for your understanding and happy coding!

Do you still need a year to reconsider?

**Comment by gileli121** [comment]
Not sure if it will help.. I developed plugin for JetBrains IDEs called "GlassIDE".
It is free and does this:
![image](https://user-images.githubusercontent.com/17680514/104809808-8c29fd00-57f8-11eb-87b8-e57ec458648e.png)
![image](https://user-images.githubusercontent.com/17680514/104809811-9a781900-57f8-11eb-84e8-41f105490d51.png)

https://github.com/gileli121/GlassIDE

I am considering if I should develop port for VSCode

**Comment by gileli121** [comment]
I developed plugin for JetBrains IDEs that allow transparency
https://www.youtube.com/watch?v=mbVLot2m6FU
I share it here in case someone want to use JetBrains instead.

Another image:
![image](https://user-images.githubusercontent.com/17680514/105539467-74bda900-5cfd-11eb-8421-d5b8aeb5f329.png)

Note that it is only for Windows 10

**Comment by convers39** [comment]
Still no update on a transparent UI solution on Mac?

**Comment by jaweii** [comment]
I developed an extension named SuperBG to set window to transparent in another way:

[![watchURL](https://raw.githubusercontent.com/jaweii/superBG/master/demo/watchURL.gif)]()

[![selectImages](https://raw.githubusercontent.com/jaweii/superBG/master/demo/selectImages.gif)]()

**Comment by sgarbesi** [comment]
@convers39 @darkguy2008 @jaweii @pcgeek86 @ckissane Made an easily injectable PR. Even if it's not accepted, it should be trivial to integrate with future upgrades to the project. Hopefully a senior coder on the project ports it or they merge the PR. Good luck guys.

https://github.com/microsoft/vscode/pull/151678

**Comment by amouratoglou** [comment]
please guys, is there any guide to make it work without those buggy plugins?
why it was not merged? thanks.

**Comment by yubaoquan** [comment]
Any progress ?

**Comment by cjwijtmans** [comment]
the issue on windows making the whole window transparent is that text becomes unreadable.

**Comment by 1-800-jono** [comment]
To make VSCode transparent (also in MacOS) install [APC Customize UI++](https://marketplace.visualstudio.com/items?itemName=drcika.apc-extension) then add this to `settings.json` file: 

```
  "apc.electron": {
    "backgroundColor": "rgba(40, 42, 54, 0.8)", // Set this to desired bg color
    "frame": false,
    "transparent": true,
    "titleBarStyle": "hiddenInset",
    "vibrancy": "ultra-dark",
    "opacity": 0.85, // Range: 0-1 (To go fully transparent, set it to 0)
    "visualEffectState": "active"
  },
  "workbench.colorCustomizations": {
    "editorGroupHeader.tabsBackground": "#00000000",
    "sideBar.background": "#00000000",
    "sideBarSectionHeader.background": "#00000000",
    "editor.background": "#00000000",
    "peekViewEditor.background": "#00000000",
    "peekViewEditorGutter.background": "#00000000",
    "peekViewTitle.background": "#00000000",
    "peekViewResult.background": "#00000000",
    "tab.activeBackground": "#00000000",
    "tab.inactiveBackground": "#00000000",
    "statusBar.background": "#00000000",
    "statusBar.debuggingBackground": "#00000000",
    "statusBar.noFolderBackground": "#00000000",
    "statusBarItem.activeBackground": "#00000000",
    "panel.background": "#00000000",
    "activityBar.background": "#00000000",
    "terminal.background": "#00000000"
  }
```

Finally :)

**Comment by AurevoirXavier** [comment]
> To make VSCode transparent (also in MacOS) install [APC Customize UI++](https://marketplace.visualstudio.com/items?itemName=drcika.apc-extension) then add this to `settings.json` file:
> 
> ```
>   "apc.electron": {
>     "backgroundColor": "rgba(40, 42, 54, 0.8)", // Set this to desired bg color
>     "frame": false,
>     "transparent": true,
>     "titleBarStyle": "hiddenInset",
>     "vibrancy": "ultra-dark",
>     "opacity": 0.85, // Range: 0-1 (To go fully transparent, set it to 0)
>     "visualEffectState": "active"
>   },
>   "workbench.colorCustomizations": {
>     "editorGroupHeader.tabsBackground": "#00000000",
>     "sideBar.background": "#00000000",
>     "sideBarSectionHeader.background": "#00000000",
>     "editor.background": "#00000000",
>     "peekViewEditor.background": "#00000000",
>     "peekViewEditorGutter.background": "#00000000",
>     "peekViewTitle.background": "#00000000",
>     "peekViewResult.background": "#00000000",
>     "tab.activeBackground": "#00000000",
>     "tab.inactiveBackground": "#00000000",
>     "statusBar.background": "#00000000",
>     "statusBar.debuggingBackground": "#00000000",
>     "statusBar.noFolderBackground": "#00000000",
>     "statusBarItem.activeBackground": "#00000000",
>     "panel.background": "#00000000",
>     "activityBar.background": "#00000000",
>     "terminal.background": "#00000000"
>   }
> ```
> 
> Finally :)

<img width="1187" alt="image" src="https://github.com/user-attachments/assets/c27dafdc-6b0f-4e53-a222-ba0b6d0b7886">

After using this, I noticed there are some ghosting effects. At first, I thought my screen was suffering from burn-in, but later I realised this only occurs in VSCode and can be captured in screenshots, indicating that there is some image caching in VSCode. How can I resolve this?

**Comment by LogixTheDev** [comment]
> To make VSCode transparent (also in MacOS) install [APC Customize UI++](https://marketplace.visualstudio.com/items?itemName=drcika.apc-extension) then add this to `settings.json` file:
> 
> ```
>   "apc.electron": {
>     "backgroundColor": "rgba(40, 42, 54, 0.8)", // Set this to desired bg color
>     "frame": false,
>     "transparent": true,
>     "titleBarStyle": "hiddenInset",
>     "vibrancy": "ultra-dark",
>     "opacity": 0.85, // Range: 0-1 (To go fully transparent, set it to 0)
>     "visualEffectState": "active"
>   },
>   "workbench.colorCustomizations": {
>     "editorGroupHeader.tabsBackground": "#00000000",
>     "sideBar.background": "#00000000",
>     "sideBarSectionHeader.background": "#00000000",
>     "editor.background": "#00000000",
>     "peekViewEditor.background": "#00000000",
>     "peekViewEditorGutter.background": "#00000000",
>     "peekViewTitle.background": "#00000000",
>     "peekViewResult.background": "#00000000",
>     "tab.activeBackground": "#00000000",
>     "tab.inactiveBackground": "#00000000",
>     "statusBar.background": "#00000000",
>     "statusBar.debuggingBackground": "#00000000",
>     "statusBar.noFolderBackground": "#00000000",
>     "statusBarItem.activeBackground": "#00000000",
>     "panel.background": "#00000000",
>     "activityBar.background": "#00000000",
>     "terminal.background": "#00000000"
>   }
> ```
> 
> Finally :)

This doesn't work for me, on Windows 11. Looks like it's a broken extension... on top of not doing anything, it throws errors:

![Image](https://github.com/user-attachments/assets/4f136052-3b63-4a9e-9e26-48ed3aa904cd)

I had more luck with [GlassIt-VSC](https://marketplace.visualstudio.com/items?itemName=s-nlf-fh.glassit). No configuration needed (though it is configurable) and no editor errors :) [Demo Screen-Rec](https://i.gyazo.com/7f6352ebf386ccf2c14f68b9a54cfd54.mp4)

It still could do with an acrylic blur though!

**Comment by Rdeisenroth** [comment]
Sooo after all these years, we're back to no working solution on Linux again... Great. Guess i'll have to go back to the `code-translucent` aur package
